src/rgw/rgw_op.cc

```diff
@@ -4493,6 +4513,34 @@ void RGWPutObj::execute(optional_yield y)
   bl.append(etag.c_str(), etag.size());
   emplace_attr(RGW_ATTR_ETAG, std::move(bl));
 
+  if (cksum_filter) {
+    const auto& hdr = cksum_filter->header();
+    auto cksum_verify =
+      cksum_filter->verify(*s->info.env); // valid or no supplied cksum
+    cksum = get<1>(cksum_verify);
+    if (std::get<0>(cksum_verify)) {
+      buffer::list cksum_bl;
+      ldpp_dout(this, 16)
+	<< fmt::format("{} content checksum verified ", hdr.second)
+	<< fmt::format("\n\tcalculated={} == \n\texpected=  {}",
+		       cksum->to_armor(),
+		       cksum_filter->expected(*s->info.env))
+	<< dendl;
```

Comment:
this dout+format pattern is showing up a lot here, and i argued against it in https://github.com/ceph/ceph/pull/56008#pullrequestreview-1944064024

both of these format() calls are going to allocate memory because the result is longer than the short string optimization. is there some reason to expect that this is still better than the style used elsewhere in ceph?
```c++
ldpp_dout(this, 16) << hdr.second << " content checksum verified\n\t"
    "calculated=" << cksum->to_armor() << " == \n\t"
    "expected=  " << cksum_filter->expected(*s->info.env))
    << dendl;
```

Reply:
> the potential difference iirc is that we expect each of these '<<' operator invocations to take locks

that's not the case, no. you might be thinking about the locale-related locking in std::ostream's constructor? in https://github.com/ceph/ceph/pull/19100 (with additional discussion in https://github.com/ceph/ceph/pull/17938) we made dout's ostream thread-local so it's only constructed once per thread and reused for each call

Reply:
I just don't think use of format to reduce << spam is an antipattern.

Reply:
i hacked together a set of dout macros for this in https://github.com/ceph/ceph/pull/58229